code/game/machinery/GBPvendor.dm

```diff
@@ -76,36 +78,30 @@
 			dat += "Rating: Holy Shit<br>"
 	dat += "</div>"
 	dat += "<br><b>Equipment point cost list:</b><BR><table border='0' width='300'>"
-	for(var/datum/data/GBP_equipment/prize in prize_list)
+	for(var/datum/GBP_equipment/prize in prize_list)
 		dat += "<tr><td>[prize.equipment_name]</td><td>[prize.cost]</td><td><A href='?src=\ref[src];purchase=\ref[prize]'>Purchase</A></td></tr>"
 	dat += "</table>"
 
 	var/datum/browser/popup = new(user, "vending", "Engineering Point Redemption", 400, 350)
 	popup.set_content(dat)
 	popup.open()
-	return
 
 /obj/machinery/GBP_vendor/Topic(href, href_list)
 	if(..())
 		return
 	if(href_list["purchase"])
-		var/datum/data/mining_equipment/prize = locate(href_list["purchase"])
+		var/datum/mining_equipment/prize = locate(href_list["purchase"])

```

Comment:
Something tells me this code wasn't working properly if I was getting it to work with this line as it was.